src/EventStore.Projections.Core/Services/Processing/ResultEventEmitter.cs

```diff
@@ -1,8 +1,11 @@
 using System;
+using System.Collections.Generic;
+using Serilog;
 
 namespace EventStore.Projections.Core.Services.Processing {
 	public class ResultEventEmitter : IResultEventEmitter {
 		private readonly ProjectionNamesBuilder _namesBuilder;
+		private List<string> WarningProjectionList = new List<string>();
```

Comment:
for non-static member vars it should be _warningProjectionList

is it possible for multiple different names to be added to the list?

Reply:
Based on the investigation and tests so far, it looks like each projection has a different ResultEventEmitter object associated with it. So in the latest changes, I have got ride of the List variable and instead introduced a bool variable to the ResultEventEmitter Class to keep track if the projection's state size is less than 8 MB